app/services/decisions/decision_tree.rb

```diff
@@ -99,25 +99,27 @@ class DecisionTree < DslDecisionTree
       .goto(edit: PRIOR_AUTHORITY_CHECK_ANSWERS)
       .goto(show: PRIOR_AUTHORITY_START_PAGE)
 
-    from(:case_contact)
-      .when(-> { application.navigation_stack[-1].match?('check_answers') })
+    # ---------------------------------
+    # prior authority application steps
+    # ---------------------------------
+    # catchall to return to the check answers page if the was the original source location
+    from(ANY)
+      .when(->(rule) { source_was_check_answers?(rule) })
       .goto(edit: PRIOR_AUTHORITY_CHECK_ANSWERS)
+
+    from(:case_contact)
       .goto(show: PRIOR_AUTHORITY_START_PAGE)
 
     from(:client_detail)
-      .when(-> { application.navigation_stack[-1].match?('check_answers') })
-      .goto(edit: PRIOR_AUTHORITY_CHECK_ANSWERS)
       .goto(show: PRIOR_AUTHORITY_START_PAGE)
 
     # prison law flow
     from(:next_hearing)
-      .when(-> { application.navigation_stack[-1].match?('check_answers') })
-      .goto(edit: PRIOR_AUTHORITY_CHECK_ANSWERS)
       .goto(show: PRIOR_AUTHORITY_START_PAGE)
 
     # non-prison law flow
     from(:case_detail)
-      .when(-> { application.navigation_stack[-1].match?('check_answers') })
+      .when(->(rule) { source_was_check_answers?(rule, any_source: true) })
```

Comment:
In theory is this rendered unnecessary by the (ANY) clause above?

Reply:
:( I wish... as normally the loop is case_details -> hearing details... 

but both these section have an edit button on the summary screen so the assumption is you exit the loop early - just for this one...